dd-sdk-android-core/src/main/kotlin/com/datadog/android/api/storage/datastore/DataStoreCallback.kt

```diff
@@ -0,0 +1,27 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2016-Present Datadog, Inc.
+ */
+
+package com.datadog.android.api.storage.datastore
+
+import com.datadog.android.core.persistence.datastore.DataStoreContent
+
+/**
+ * Callback for asynchronous operations on the datastore.
+ */
+interface DataStoreCallback<T : Any> {
+
+    /**
+     * Triggered on successfully fetching data from the datastore.
+     *
+     * @param dataStoreContent contains the datastore content if there was data to fetch.
+     */
+    fun onSuccess(dataStoreContent: DataStoreContent<T>?)
+
+    /**
+     * Triggered when an exception occurred getting data from the datastore.
```

Comment:
nit: maybe we should say that it is triggered when we cannot get data? we say it is triggered on exception, but there is no exception in the argument.